mods/default/functions.lua

```diff
@@ -585,7 +585,8 @@ function default.can_interact_with_node(player, pos)
 
 	-- Is player wielding the right key?
 	local item = player:get_wielded_item()
-	if item:get_name() == "default:key" then
+	local def = minetest.registered_items[item:get_name()]
+	if def and def.groups and def.groups.key == 1 then
```

Comment:
Using `item:get_definition()` or `minetest.get_item_group(item:get_name(), "key")` might be better, idk.

Reply:
Do you need to check if group key exist then?

Reply:
With `local def = item:get_definition()`, you don't need to check for `def` but still for `def.groups`. (Instead of `nil`, there's a fallback def[.](https://github.com/minetest/minetest/blob/eef62c82a2e58700fc1216b0b8c03e421bc77995/src/script/lua_api/l_item.cpp#L282))
[`minetest.get_item_group`](https://github.com/minetest/minetest/blob/ade7a1cbcf7156ebddb12bd2ec2ea60818d72b19/builtin/game/misc.lua#L146) always gives you a number.

Reply:
Changed.